src/call_builder.ts

```diff
@@ -51,12 +70,28 @@ export class CallBuilder {
    * Triggers a HTTP request using this builder's current configuration.
    * @returns {Promise} a Promise that resolves to the server's response.
    */
-  call() {
+  public call(): Promise<T> {
     this.checkFilter();
     return this._sendNormalRequest(this.url).then((r) =>
       this._parseResponse(r)
     );
   }
+  //// TODO: Migrate to async, BUT that's a change in behavior and tests "rejects two filters" will fail.
+  //// It's because async will check within promise, which makes more sense when using awaits instead of Promises.
+  // public async call(): Promise<T> {
+  //   this.checkFilter();
+  //   const r = await this._sendNormalRequest(this.url);
+  //   return this._parseResponse(r);
+  // }
+  //// /* actually equals */
+  //// public call(): Promise<T> {
+  ////   return Promise.resolve().then(() => {
+  ////     this.checkFilter();
+  ////     return this._sendNormalRequest(this.url)
+  ////   }).then((r) => {
+  ////     this._parseResponse(r)
+  ////   });
+  //// }
```

Comment:
I'm might not be 100% understanding what you're conveying, but generally I'd like to keep functionality exactly the same unless there's a compelling reason not to (for example, it really goes against user expectations, or there's a strong performance improvement). I don't see either of these features in your new code, but please let me know if I'm mistaken!

Reply:
Created new ticket https://github.com/stellar/js-stellar-sdk/issues/296